bin/load-spark-env.sh

```diff
@@ -20,7 +20,29 @@
 # This script loads spark-env.sh if it exists, and ensures it is only loaded once.
 # spark-env.sh is loaded from SPARK_CONF_DIR if set, or within the current directory's
 # conf/ subdirectory.
-FWDIR="$(cd "`dirname "$0"`"/..; pwd)"
+realpath () {
```

Comment:
Can this function be replaced by:

    DIR="$(dirname "$(readlink -f "$0")")"

I see you're recursively resolving the symlink and putting a limit in the depth of recursion, but it sounds like `readlink -f` does just that.